### DISCUSSION TITLE
Why does Fastapi need more time when use CryptContext？

### DISCUSSION THREAD

**Discussion by ssrshz** [fastapi/fastapi#6193] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
==================Fastapi==========================
from passlib.context import CryptContext
import time
from loguru import logger

pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")


def verify(plain_password, hashed_password):
    start = time.perf_counter_ns()
    verify_result = pwd_context.verify(plain_password, hashed_password)
    end = time.perf_counter_ns()
    logger.info(f"verify end at {(end - start) / (10 ** 9)}s")
    return verify_result

@user_router.get("/test")
async def test():
    plain_password = '12345'
    hashed_password = "$2b$12$9QqLfn..1PGFMLtoD1xvFOX8IGk/COLBHi8yKLFs.DTsiCyi9QmYq"
    result = verify(plain_password, hashed_password)

    return {"message": result}

--------------------result------------------------
2022-11-16 17:27:45.874 | INFO     | verify:12 - verify end at 0.280580125s
INFO:     127.0.0.1:51879 - "GET /user/test HTTP/1.1" 200 OK


=====================Python=========================

from loguru import logger
import time
from passlib.context import CryptContext

hashed_password = "$2a$10$7LwEZM8ylt6NJPGVkZZ4euMNHL68dsD3gOe9/9A69cIbjuerVV6wS"
plain_password = "12345"


v_start = time.perf_counter_ns()
pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
verify_result = pwd_context.verify(plain_password, hashed_password)
v_end = time.perf_counter_ns()
logger.info(f"verify end at {(v_end - v_start) / (10 ** 9)}s")
--------------------result------------------------
2022-11-16 17:30:35.476 | INFO     | __main__:<module>:14 - verify end at 0.087431334s
```

### Description

Fastapi speeds 0.280580125s
Python speeds 0.087431334s

why???

### Operating System

macOS

### Operating System Details

Macos Ventura 13.0 

### FastAPI Version

0.85.1

### Python Version

Python 3.10.8

**Comment by yinziyan1206** [comment, marked as answer]
`$2b$` is slower than `$2a$` 
![image](https://user-images.githubusercontent.com/37829370/202334040-340942e7-c3cb-4bec-bb00-f12e939b9d9d.png)
![image](https://user-images.githubusercontent.com/37829370/202334068-6e3d0b5f-344c-41e5-ad4a-00da46342179.png)

**Comment by ssrshz** [comment]
@yinziyan1206 Thanks a lot，i'v got confused for a long time.

**Comment by tiangolo** [comment]
Thanks for the help here @ycd ! 👏  🙇 

Thanks for reporting back and closing the issue  @ssrshz 👍
